### PR TITLE
Run deduplicate.sh as cron job

### LINKED ISSUE #165: rdfind in deduplicate.sh is getting slow, make it a cron job
It takes several minutes to run on the full wikis folder, and all it does is free up some disk space.

### PR DESCRIPTION
**Pull request by edg2s**
Fixes #165

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ Tested on Ubuntu 18.04 and Debian 10.
 
 * Put all these files in `/var/www/html`
 * Run `sudo setup.sh`
+* Create a daily cron job as to run `deduplicate.sh` using the same account as your webserver
```
Comment (MatmaRex) [inline README.md:24]: The command you added to setup.sh already does this, right? I wouldn't mention it here.

File: setup.sh [hunk 2]
```diff
@@ -28,6 +27,10 @@ GRANT ALL PRIVILEGES ON \`patchdemo\_%\`.* TO 'patchdemo'@'localhost';
 composer update --no-dev
 npm install --production
 
+# setup daily cron job to deduplicate files
+BASEDIR=$(dirname "$0")
+sudo ln -s $BASEDIR/deduplicate.sh /etc/cron.daily/
```
Comment (MatmaRex) [inline setup.sh:32]: I didn't know jobs can be scheduled this way, neat.

Comment (edg2s) [reply setup.sh:32]: Neither did I until now.

File: README.md (hunk 1 above)
Comment (edg2s) [reply README.md:24]: Oh yeah, I fixed it later

**Comment by MatmaRex** [comment]
I ran the setup script on patchdemo.wmflabs.org, so it should start running daily. But I don't trust cron jobs, so to verify, I just created two identical test wikis and checked their size.
```
matmarex@patchdemo:/var/www/html$ du -sh wikis/c3058b39da82d3c238a3ab3972ded2cc wikis/6f7b4fe31f8353e49cb49f5488cd3807
392M    wikis/c3058b39da82d3c238a3ab3972ded2cc
392M    wikis/6f7b4fe31f8353e49cb49f5488cd3807
```
I am expecting one of them to drop to ~0 after the deduplicate job runs.

**Comment by MatmaRex** [comment]
It does not seem to be working. Those two directories are still not deduplicated.

**Comment by edg2s** [comment]
What if you run the script manually (as root from the root folder)? Does it have execute permissions?

**Comment by edg2s** [comment]
Looks like the symlink is not set up correctly:
image

**Comment by edg2s** [comment]
Let's see if this works tomorrow:
image

**Comment by edg2s** [comment]
I ran `du -sh` on a few wikis created earlier in October before this patch, and they all seem to be ~390M:
image

**Comment by MatmaRex** [comment]
It didn't work still. I'm trying to look into it.

**Comment by MatmaRex** [comment]
The deduplicate script doesn't actually support being called through a symlink. It will try to deduplicate /etc/cron.daily/wikis/ and do nothing. This might help:
```
BASEDIR=$(dirname $(readlink -f "$0"))
```
But more seriously, Cron refuses to run it at all, because of this: https://stackoverflow.com/a/43454485
> the files under these directories have to be pass some sanity checks including the following: be executable, be owned by root, not be writable by group or other and, if symlinks, point to files owned by root. Additionally, the file names must conform to the filename requirements of run-parts: they must be entirely made up of letters, digits and can only contain the special signs underscores ('_') and hyphens ('-'). Any file that does not conform to these requirements will not be executed by run-parts. For example, any file containing dots will be ignored.

**Pull request merged**